Makefile

```diff
@@ -2,26 +2,50 @@ HOSTNAME=github.com
 NAMESPACE=iterative
 NAME=iterative
 VERSION=0.0.${shell date +%s}+development
-OS_ARCH=${shell go env GOOS}_${shell go env GOARCH}
-BINARY=terraform-provider-${NAME}
-INSTALL_PATH=~/.terraform.d/plugins/${HOSTNAME}/${NAMESPACE}/${NAME}/${VERSION}/${OS_ARCH}
+GOOS=${shell go env GOOS}
+GOARCH=${shell go env GOARCH}
+TF_PLUGIN_INSTALL_PATH=~/.terraform.d/plugins/${HOSTNAME}/${NAMESPACE}/${NAME}/${VERSION}/${GOOS}_${GOARCH}
+TPI_PATH ?= $(shell pwd)
+LEO_PATH ?= $(shell pwd)/cmd/leo
+GO_LINK_FLAGS ?= -s -w -X terraform-provider-iterative/iterative/utils.Version=${VERSION}
+
+LEO_BUILD_COMMAND ?= \
+	CGO_ENABLED=0 \
+	go build -ldflags="$(GO_LINK_FLAGS)" \
+	-o $(shell pwd)/leo-$(GOOS)-$(GOARCH) \
+	$(LEO_PATH)
 
 default: build
 
+# TODO: Should this be renamed to something like build-check ?
+.PHONY: build
 build:
 	go build ./...
 
+.PHONY: leo-bin
+leo-bin:
+	$(LEO_BUILD_COMMAND)
+
+.PHONY: install_tpi
 install_tpi:
```

Comment:
`-Wunderscore-target-name`

Reply:
I'd prefer `snake-case` if necessary or single words otherwise. E.g. if `leo` is enough, don't use `leo-bin` instead?

@iterative/cml, feel free to `(up|down)vote` this comment.

Reply:
Coup de main at https://github.com/iterative/terraform-provider-iterative/pull/701#pullrequestreview-1159029861

Reply:
@0x2b3bfa0 
> `snake-case`

you mean `snake_case` or `kebab-case` ? 😵

Reply:
Edited. 🤦‍♂️

`<kidding>`
I clearly meant `kebabCase`
`</kidding>`